packages/@uppy/core/src/Uppy.js

```diff
@@ -434,6 +434,18 @@ class Uppy {
    * The `files` value is passed in because it may be updated by the caller without updating the store.
    */
   #checkAndCreateFileStateObject (files, fileDescriptor) {
+    // Uppy expects files in { name, type, size, data } format.
+    // If the actual File object is passed from input[type=file] or drag-drop,
+    // we normalize it to match Uppy file object
+    if (!fileDescriptor.data && !fileDescriptor.isRemote) {
```

Comment:
This seems like a indirect way to guess that we are dealing with a `File`. Wouldn't it be better to do:
```suggestion
    if (fileDescriptor instanceof File) {
```

Reply:
Good idea! I suspect there might be issue with this detection failing for some weird reason, but let’s try. @aduh95 do you see any potential problems?